src/Akka.Streams.Kafka/Stages/CommitableConsumerStage.cs

```diff
@@ -0,0 +1,178 @@
+            });
+        }
+
+        public override void PreStart()
+        {
+            base.PreStart();
+
+            _consumer = _settings.CreateKafkaConsumer();
+            _consumer.OnMessage += HandleOnMessage;
+            _consumer.OnConsumeError += HandleConsumeError;
+            _consumer.OnError += HandleOnError;
+            _consumer.OnPartitionsAssigned += HandleOnPartitionsAssigned;
+            _consumer.OnPartitionsRevoked += HandleOnPartitionsRevoked;
+
+            switch (_subscription)
+            {
+                case TopicSubscription ts:
+                    _consumer.Subscribe(ts.Topics);
+                    break;
+                case Assignment a:
+                    _consumer.Assign(a.TopicPartitions);
+                    break;
+                case AssignmentWithOffset awo:
+                    _consumer.Assign(awo.TopicPartitions);
+                    break;
+            }
+
+            _messagesReceived = GetAsyncCallback<Message<K, V>>(MessagesReceived);
+            _partitionsAssigned = GetAsyncCallback<IEnumerable<TopicPartition>>(PartitionsAssigned);
+            _partitionsRevoked = GetAsyncCallback<IEnumerable<TopicPartition>>(PartitionsRevoked);
+        }
+
+        public override void PostStop()
+        {
+            _consumer.OnMessage -= HandleOnMessage;
+            _consumer.OnConsumeError -= HandleConsumeError;
+            _consumer.OnError -= HandleOnError;
+            _consumer.OnPartitionsAssigned -= HandleOnPartitionsAssigned;
+            _consumer.OnPartitionsRevoked -= HandleOnPartitionsRevoked;
+
+            _consumer.Dispose();
+
+            base.PostStop();
+        }
+
+        //
+        // Consumer's events
+        //
+
+        private void HandleOnMessage(object sender, Message<K, V> message) => _messagesReceived.Invoke(message);
+
+        // TODO: how I should react?
+        private void HandleConsumeError(object sender, Message message) { }
+
+        private void HandleOnError(object sender, Error error)
+        {
+            Log.Error(error.Reason);
+
+            if (!KafkaExtensions.IsBrokerErrorRetriable(error) && !KafkaExtensions.IsLocalErrorRetriable(error))
+            {
+                FailStage(new Exception(error.Reason));
+            }
+        }
+
+        private void HandleOnPartitionsAssigned(object sender, List<TopicPartition> list)
+        {
+            _partitionsAssigned.Invoke(list);
+        }
+
+        private void HandleOnPartitionsRevoked(object sender, List<TopicPartition> list)
+        {
+            _partitionsRevoked.Invoke(list);
+        }
+
+        //
+        // Async callbacks
+        //
+
+        private void MessagesReceived(Message<K, V> message)
+        {
+            var consumer = _consumer;
+            var commitableOffset = new CommitableOffset(
+                () => consumer.CommitAsync(),
+                new PartitionOffset("groupId", message.Topic, message.Partition, message.Offset));
```

Comment:
I should pass groupID here

PartitionOffset could be redundant. I could pass `Message<K, V>` instead